src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -369,6 +369,89 @@ def translate_literal(
     return [(data_node, data_type)]
 
 
+def translate_scalar_expr(
+    node: gtir.Node,
+    sdfg: dace.SDFG,
+    state: dace.SDFGState,
+    sdfg_builder: gtir_to_sdfg.SDFGBuilder,
+    reduce_identity: Optional[gtir_to_tasklet.SymbolExpr],
+) -> list[TemporaryData]:
+    assert isinstance(node, gtir.FunCall)
+    assert isinstance(node.type, ts.ScalarType)
+
+    args = []
+    connectors = []
+    scalar_expr_args = []
+
+    for arg_expr in node.args:
+        is_symbol = True
+        if isinstance(arg_expr, gtir.SymRef):
+            try:
+                sdfg_builder.get_symbol_type(arg_expr.id)
+            except KeyError:
+                is_symbol = False
+
+        if is_symbol:
+            # we visit the argument expression and obtain the access node to
+            # a scalar data container, which will be connected to the tasklet
+            arg_node, arg_type = sdfg_builder.visit(
+                arg_expr,
+                sdfg=sdfg,
+                head_state=state,
+                reduce_identity=reduce_identity,
+            )[0]
+            if not (
+                isinstance(arg_type, ts.ScalarType)
+                and isinstance(arg_node.desc(sdfg), dace.data.Scalar)
+            ):
+                raise ValueError(f"Invalid argument to scalar expression {arg_expr}.")
+            param = f"__in_{arg_node.data}"
+            args.append(arg_node)
+            connectors.append(param)
+            scalar_expr_args.append(gtir.SymRef(id=param))
+        else:
+            # this is the case of reverved symbols, e.g. 'float64' used as type argument to `cast_`
```

Comment:
```suggestion
            # this is the case of reserved symbols, e.g. 'float64', used as type argument to `cast_`
```

I think the distinction here is quite critical and important. I would rephrase it to something like "non variable argument, e.g. target type, such as `float64`, in a casting expression, `cast_(vaiable, float64)`."
To make it more clear what you do here.

Reply:
Good point.